sklearn/discriminant_analysis.py

```diff
@@ -181,6 +299,22 @@ class LinearDiscriminantAnalysis(BaseEstimator, LinearClassifierMixin,
 
         .. versionadded:: 0.17
 
+    covariance_estimator : estimator, default=None
+        The covariance estimator is a BaseEstimator with a fit
```

Comment:
May I propose:

> If not None, `covariance_estimator` is used to estimate the covariance matrices instead of relying the empirical covariance estimator (with potential shrinkage). The object should have a fit method and a covariance_ attribute like the estimators in sklearn.covariance